src/client/Client.h

```diff
@@ -405,34 +412,54 @@ class Client : public Dispatcher, public md_config_obs_t {
   int fsetattrx(int fd, struct ceph_statx *stx, int mask, const UserPerm& perms);
   int chmod(const char *path, mode_t mode, const UserPerm& perms);
   int fchmod(int fd, mode_t mode, const UserPerm& perms);
+  int fchmodat(int dirfd, const char *relpath, mode_t mode, int flags, const UserPerm& perms);
   int lchmod(const char *path, mode_t mode, const UserPerm& perms);
   int chown(const char *path, uid_t new_uid, gid_t new_gid,
 	    const UserPerm& perms);
   int fchown(int fd, uid_t new_uid, gid_t new_gid, const UserPerm& perms);
   int lchown(const char *path, uid_t new_uid, gid_t new_gid,
 	     const UserPerm& perms);
+  int fchownat(int dirfd, const char *relpath, uid_t new_uid, gid_t new_gid,
+               int flags, const UserPerm& perms);
   int utime(const char *path, struct utimbuf *buf, const UserPerm& perms);
   int lutime(const char *path, struct utimbuf *buf, const UserPerm& perms);
   int futime(int fd, struct utimbuf *buf, const UserPerm& perms);
   int utimes(const char *relpath, struct timeval times[2], const UserPerm& perms);
   int lutimes(const char *relpath, struct timeval times[2], const UserPerm& perms);
   int futimes(int fd, struct timeval times[2], const UserPerm& perms);
   int futimens(int fd, struct timespec times[2], const UserPerm& perms);
+  int utimensat(int dirfd, const char *relpath, struct timespec times[2], int flags,
+                const UserPerm& perms);
   int flock(int fd, int operation, uint64_t owner);
   int truncate(const char *path, loff_t size, const UserPerm& perms);
 
   // file ops
   int mknod(const char *path, mode_t mode, const UserPerm& perms, dev_t rdev=0);
+
+  int create_and_open(boost::optional<int> dirfd, const char *relpath, int flags, const UserPerm& perms,
```

Comment:
```suggestion
  int create_and_open(std::optional<int> dirfd, const char *relpath, int flags, const UserPerm& perms,
```

Reply:
Hmm... any specific reason?

Reply:
I would think we'd prefer the `std` version, would we not?

Reply:
fair enough